services/galley/test/integration/API/MLS.hs

```diff
@@ -2381,13 +2381,63 @@ testJoinSubConv = do
         createExternalCommit alice1 Nothing (fmap (flip SubConv subId) qcnv)
           >>= sendAndConsumeCommitBundle
 
--- FUTUREWORK: implement the following tests
-
 testJoinSubNonMemberClient :: TestM ()
-testJoinSubNonMemberClient = pure ()
+testJoinSubNonMemberClient = do
+  [alice, bob] <- createAndConnectUsers [Nothing, Nothing]
+
+  runMLSTest $ do
+    [alice1, alice2, bob1] <-
+      traverse createMLSClient [alice, alice, bob]
+    traverse_ uploadNewKeyPackage [bob1, alice2]
+    (_, qcnv) <- setupMLSGroup alice1
+    void $ createAddCommit alice1 [alice] >>= sendAndConsumeCommit
+
+    let subId = SubConvId "conference"
+    void $ createSubConv qcnv alice1 subId
+
+    -- now Bob attempts to get the group info so he can join via external commit
+    -- with his own client, but he cannot because he is not a member of the
+    -- parent conversation
+    getGroupInfo (ciUser bob1) (fmap (flip SubConv subId) qcnv)
+      !!! const 404 === statusCode
 
 testAddClientSubConv :: TestM ()
-testAddClientSubConv = pure ()
+testAddClientSubConv = do
+  [alice, bob] <- createAndConnectUsers [Nothing, Nothing]
+  runMLSTest $ do
+    [alice1, bob1] <- traverse createMLSClient [alice, bob]
+    void $ uploadNewKeyPackage bob1
+    (_, qcnv) <- setupMLSGroup alice1
+    void $ createAddCommit alice1 [bob] >>= sendAndConsumeCommit
+
+    let subId = SubConvId "conference"
+    void $ createSubConv qcnv alice1 subId
+
+    void $ uploadNewKeyPackage bob1
+
+    commit <- createAddCommit alice1 [bob]
+    (createBundle commit >>= postCommitBundle (mpSender commit))
+      !!! do
+        const 400 === statusCode
+        const (Just "Add proposals in subconversations are not supported")
+          === fmap Wai.message . responseJsonError
+
+    finalSub <-
+      liftTest $
+        responseJsonError
+          =<< getSubConv (qUnqualified alice) qcnv subId
+            <!! const 200 === statusCode
+    liftIO $ do
+      assertEqual
+        "The subconversation has Bob in it, while it shouldn't"
+        [alice1]
+        (pscMembers finalSub)
+      assertEqual
+        "The subconversation epoch has moved beyond 1"
+        (Epoch 1)
+        (pscEpoch finalSub)
+
+-- FUTUREWORK: implement the following tests
```

Comment:
I think having these stub tests below feels ripe for both merge conflicts and the assumption of tests existing by their outputs being green instead of yellow or something like that,.